pkgs/applications/networking/browsers/links2/default.nix

```
@@ -1,23 +1,23 @@
 { lib, stdenv, fetchurl
 , gpm, openssl, pkg-config, libev # Misc.
-, libpng, libjpeg, libtiff, librsvg # graphic formats
+, libpng, libjpeg, libtiff, librsvg, libavif # graphic formats
 , bzip2, zlib, xz # Transfer encodings
-, enableFB ? true
+, enableFB ? (!stdenv.isDarwin)
 , enableDirectFB ? false, directfb
-, enableX11 ? true, libX11, libXt, libXau # GUI support
+, enableX11 ? (!stdenv.isDarwin), libX11, libXt, libXau # GUI support
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   version = "2.29";
   pname = "links2";
 
   src = fetchurl {
-    url = "${meta.homepage}/download/links-${version}.tar.bz2";
-    sha256 = "sha256-IqqWwLOOGm+PftnXpBZ6R/w3JGCXdZ72BZ7Pj56teZg=";
+    url = "${finalAttrs.meta.homepage}/download/links-${finalAttrs.version}.tar.bz2";
```

```suggestion
    url = "http://links.twibright.com/download/links-${finalAttrs.version}.tar.bz2";
```
using meta.homepage here is not very correct or future proof. If meta.homepage gets changed in the future this will likely be forgotten

---

That makes sense, @SuperSandro, is this important enough to you, that you'd accept a PR for it or it is something that should rather be addressed with the next update to the package?

---

feel free to make PR now, me or @SuperSandro2000 happy to have a look at it

---

Sure thing, @kirillrdy, here's the PR: #289130

---

> If meta.homepage gets changed in the future this will likely be forgotten

Note that if the domain changes in the future, the URL 100% needs changing, but there is a chance that the URL structure is the same, so I am not sure how this cleanup can _ever_ be useful.

---

this is similar to https://github.com/NixOS/nixpkgs/issues/277994

for links2 its most likely that url and homepage will go in sync, but that's not the case for a lot of other derivations

eg homepage can change to be different domain but src remains on github, or sometimes they go backwards